crates/cmd/src/report.rs

```diff
@@ -0,0 +1,206 @@
+use clap::ValueEnum;
+use std::path::PathBuf;
+use std::process::{ExitCode, Termination};
+use wardstone::key;
+use wardstone::primitive::asymmetric::Asymmetric;
+use wardstone_core::primitive::hash::Hash;
+
+use crate::Verbosity;
+
+const INDENT: &str = "         ";
+
+pub(crate) enum GoodPath {
+  Hash(Hash, Hash),
+  SigAlg(Asymmetric, Asymmetric),
+}
+
+impl GoodPath {
+  fn display(&self, format: ReportFormat) -> String {
+    match format {
+      ReportFormat::Human => match self {
+        GoodPath::Hash(got, want) => {
+          format!("{INDENT}OK hash function(got: {got}, want: {want})")
+        },
+        GoodPath::SigAlg(got, want) => {
+          format!("{INDENT}OK signature algorithm(got: {got}, want: {want})")
+        },
+      },
+      ReportFormat::Json => match self {
+        GoodPath::Hash(got, want) => {
+          format!(r#""hash_function": {{ "passed": true, "got": "{got}", "want": "{want}" }}"#)
+        },
+        GoodPath::SigAlg(got, want) => {
+          format!(
+            r#""signature_algorithm": {{ "passed": true, "got": "{got}", "want": "{want}" }}"#
+          )
+        },
+      },
+    }
+  }
+}
+
+pub(crate) enum BadPath {
+  ReadError(key::Error),
+  MismatchedHash(Hash, Hash),
+  MismatchedSigAlg(Asymmetric, Asymmetric),
```

Comment:
Instead of mismatch, perhaps we could clarify that the primitives were noncompliant here.

```suggestion
  NoncompliantHash(Hash, Hash),
  NoncompliantSignatureAlgorithm(Asymmetric, Asymmetric),
```

Reply:
`NoncompliantSignatureAlgorithm` was _pretty_ long, so I shortened it to `NoncompliantSignatureAlg`. I can also make the full version, if you want, though.

Reply:
The short one is fine too.